### ISSUE TITLE
Create a toolset to make Native modules easier to write

### ISSUE THREAD

**Issue by wende** [wende/elmchemy#135]
### Internal use:
  

- [x] curry/1 macro - Copies a module function and defines it in a curried version

### External use:
  

- [x] defswap/2 macro - Defines a function head that delegates to another module with last argument being first ( `a, b, c` becomes `c, a, b`)
   ```elixir
   defswap foldl/3, to: List.foldl
   ### being equivalent of
   def foldl(x1, x2, x3), do: List.foldl(x3, x1, x2)
    ```
- [x] defreverse/2 macro - Defines a function head that delegates to another module with all arguments in reversed order (`a, b, c` becomes `c, b, a`)
 ```elixir
   defreverse foldl/3, to: List.foldl
   ### being equivalent of
   def foldl(x1, x2, x3), do: List.foldl(x3, x2, x1)
    ```

**Comment by wende** [comment]
@Hajto Preferably the solution should be a part of wende/elmchemy-core instead, since it contains all elixir helper files anyway

**Comment by wende** [comment]
Closed by wende/elmchemy-core#2
